ml/src/main/scala/org/apache/spark/ml/h2o/models/H2OMOJOPipelineModel.scala

```diff
@@ -149,8 +149,11 @@ class H2OMOJOPipelineModel(override val uid: String) extends H2OMOJOModelBase[H2
 
       // Transform the resulted Array of predictions into own but temporary columns
       // Temporary columns are created as we can't create the columns directly as nested ones
-      val predictionCols = $(outputCols).indices.map(idx => selectFromArray(idx)(frameWithPredictions.col(getPredictionCol() + ".preds")))
-      val frameWithExtractedPredictions = frameWithPredictions.withColumns(tempColNames, predictionCols)
+      var frameWithExtractedPredictions: DataFrame = frameWithPredictions
```

Comment:
nit: Would it be possible to avoid using `var` variables? Maybe via `foldLeft`?

Reply:
Thanks, will do these improvements in the next PR

Reply:
Good one!